jax/interpreters/pxla.py

```diff
@@ -681,32 +681,45 @@ def dynamic_fun(dummy, *args):
     # XLA computation at all; we handle this as a special case so we can stage
     # out multi-replica XLA computations regardless of the hardware available.
     # The 'None' values here are just dummies we know will be ignored.
-    handlers = [_pval_to_result_handler(axis_size, None, pval, local_devices,
-                                        backend)
-                for pval in out_pvals]
+    handlers = [
+        _pval_to_result_handler(axis_size, None, None, None, pval, local_devices,
+                                backend) for pval in out_pvals
+    ]
     results = [handler(None) for handler in handlers]
     return lambda *_: results
 
   jaxpr_replicas = xla.jaxpr_replicas(jaxpr)
   num_local_replicas = axis_size * jaxpr_replicas
```

Comment:
This is one of the things that needs to change in order to support multihost pmap of sharded jit?

Reply:
No this part works! It should just be doing something equivalent to this diff hunk: https://github.com/google/jax/pull/3107/files#diff-6609803133099eb146acd9d3128dccc3R624-R632, but ideally in a way that still raises the original error when appropriate. I think we just need to move that logic to after we've traced the jaxpr.